lib/galaxy/webapps/galaxy/fast_app.py

```diff
@@ -5,7 +5,24 @@
 from starlette.responses import Response
 
 from galaxy.exceptions import MessageException
-from galaxy.web.framework.decorators import api_error_message, validation_error_to_message_exception
+from galaxy.web.framework.decorators import (
+    api_error_message,
+    validation_error_to_message_exception
+)
+
+# https://fastapi.tiangolo.com/tutorial/metadata/#metadata-for-tags
+api_tags_metadata = [
+    {
+        "name": "licenses",
+        "description": "Operations with [SPDX licenses](https://spdx.org/licenses/).",
+    },
+]
+
+
+def add_open_api_tags_metadata(app: FastAPI) -> None:
+    """Adds OpenAPI documentation to the APIRouter tags"""
+    if app:
+        app.openapi_tags = api_tags_metadata
```

Comment:
Could we just add this on line 52 as the docs do (`FastAPI(openapi_tags= api_tags_metadata)`)? I'm not sure setting the attribute on the instance is really the public API we should be using?

Reply:
Yes, that was my first approach, but then I noticed that when launching from `/scripts/fapi.py` the app was already instanced in `fapi.py` [line 12](https://github.com/galaxyproject/galaxy/blob/dev/scripts/fapi.py#L12). I am not sure if this script will be just temporal and we assume that the only creation point will be on line 52.

Reply:
I think we can do this:
```
diff --git a/scripts/fapi.py b/scripts/fapi.py
index c24d54e9bf..1d94458bce 100644
--- a/scripts/fapi.py
+++ b/scripts/fapi.py
@@ -1,5 +1,4 @@
 import uvicorn
-from fastapi import FastAPI
 from galaxy_main import (
     GalaxyConfigBuilder,
     main,
@@ -9,16 +8,13 @@ from galaxy.util.properties import load_app_properties
 from galaxy.webapps.galaxy.buildapp import app_factory
 from galaxy.webapps.galaxy.fast_app import initialize_fast_app
 
-app = FastAPI()
-
 
 def wsgiloop(args, log):
     config_builder = GalaxyConfigBuilder(args)
     kwds = config_builder.app_kwds()
     kwds = load_app_properties(**kwds)
     gx = app_factory(global_conf=config_builder.global_conf(), **kwds)
-    initialize_fast_app(gx, app=app)
-    uvicorn.run(app)
+    uvicorn.run(initialize_fast_app(gx))
 
 
 if __name__ == '__main__':

```
I only created the app object in the global namespace so we could use gunicorn, but that doesn't work yet with the way we the configuration is read.

Reply:
I'll open a separate PR for that so we can keep this PR as a clean example for how to do migrations.

Reply:
Cool! I'll remove the `add_open_api_tags_metadata` then :)